front/backup.php

```diff
@@ -62,65 +62,50 @@
 
 
 /**
- * Genere un fichier backup.xml a partir de base dbhost connecte avec l'utilisateur dbuser
- * et le mot de passe dbpassword sur le serveur dbdefault
-**/
+ * Generate an XML backup file of the database
+ * @global DB $DB
+ * @global string $path Backup storage path
+ */
 function xmlbackup() {
-   global $CFG_GLPI, $DB;
-
-   //on parcoure la DB et on liste tous les noms des tables dans $table
-   //on incremente $query[] de "select * from $table"  pour chaque occurence de $table
-
-   $result = $DB->list_tables();
-   $i      = 0;
-   while ($line = $DB->fetch_row($result)) {
-      $table = $line[0];
-
-      $query[$i] = "SELECT *
-                    FROM `$table`";
-      $i++;
+   global $DB, $path;
+
+   // Will contain the select * from $table
+   $query = [];
+   // Foreach DB tables...
+   foreach ($DB->listTables() as $id => $row) {
+      // Save the query...
+      $query[$id] = "SELECT * FROM `" . $row['TABLE_NAME'] . "`";
    }
 
-   //le nom du fichier a generer...
-   //Si fichier existe deja il sera remplac�par le nouveau
-   $chemin = GLPI_DUMP_DIR."/backup.xml";
+   // Filename
+   $time_file = date("Y-m-d-H-i");
+   $filename = $path . "/glpi-" . GLPI_VERSION . "-$time_file.xml";
```

Comment:
Maybe "glpi-backup-..."?

Reply:
I'd vote to keep the `GLPI_DUMP_DIR` constant rather than a global.

Reply:
@trasher : indeed, I re-used the [SQL dump code](https://github.com/glpi-project/glpi/blob/master/front/backup.php#L427) here.

+1 for using the global, I will update the SQL  code too !